aws/route_table.go

```diff
@@ -97,29 +97,44 @@ func (t *RouteTable) ListPossibleVips() *MaybeVips {
 	return &MaybeVips{t, ids, vips, names}
 }
 
-func (t *RouteTable) GetSrcInstanceByVip(vip string) (*Ec2Meta, error) {
+func (t *RouteTable) GetSrcByVip(vip string) (*Ec2Meta, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeOut)
 	defer cancel()
 
-	id, err := t.getSrcInstanceIdByVip(vip)
+	id, err := t.getSrcByVip(vip)
 	if err != nil {
 		return nil, err
 	}
-	name, err := t.cli.getInstanceNameById(ctx, id)
-	if err != nil {
-		return nil, err
+
+	var name string
+	switch {
+	case strings.HasPrefix(id, "i-"):
+		name, err = t.cli.getInstanceNameById(ctx, id)
+		if err != nil {
+			return nil, err
+		}
+	case strings.HasPrefix(id, "eni-"):
+		name, err = t.cli.getENINameById(ctx, id)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I don't know if there are other cases, but is it better to have a `default` statement?